File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // User prompt
    string text = get_string("Text: ");

    // Check length of a string
    int i = strlen(text);

    int letters = 0;
    int words = 0;
    int sentences = 0;

    // Loop for counting
    for (int x = 0; x < i; x++)
    {
        // Count letters
        char c = text[x];
        if (isalpha(c) != 0)
        {
            letters++;
        }

        // Count words
        if (c == ' ')
        {
            words++;
        }

        // Count Sentences
        if (c == '.' || c == '!' || c == '?')
        {
            sentences++;
        }
    }

    // Account for last word
    words = words + 1;

    // Calculate the readability
    float L = ((float)letters / (float)words) * 100;
    float s = ((float)sentences / (float)words) * 100;
    float subindex = 0.0588 * L - 0.296 * s - 15.8;
    int index = round(subindex);
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: sentimental-readability/readability.py
# declaration of variables
count_letter = 0
count_word = 1
count_sentence = 0

# user inpit
text = input("Text: ")
text_length = len(text)

# count letters
for i in range(text_length):
    if (text[i].isalpha()):
        count_letter += 1

# count words
    if (text[i].isspace()):
        count_word += 1

# count sentences
    if (text[i] == '.' or text[i] == '?' or text[i] == '!'):
        count_sentence += 1

# calculate Index using formula
calculation = (0.0588 * count_letter / count_word * 100) - (0.296 * count_sentence / count_word * 100) - 15.8

index = round(calculation)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")
